HLTrigger/special/src/HLTHcalCalibTypeFilter.cc

```diff
@@ -82,35 +83,57 @@ HLTHcalCalibTypeFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::Eve
   
   edm::Handle<FEDRawDataCollection> rawdata;  
   iEvent.getByToken(DataInputToken_,rawdata);
-  
-  // checking FEDs for calibration information
-  int calibType = -1 ; int numEmptyFEDs = 0 ; 
-  std::vector<int> calibTypeCounter(8,0) ;
-  for (int i=FEDNumbering::MINHCALFEDID;
-       i<=FEDNumbering::MAXHCALFEDID; i++) {
-      const FEDRawData& fedData = rawdata->FEDData(i) ; 
-      if ( fedData.size() < 24 ) numEmptyFEDs++ ; 
-      if ( fedData.size() < 24 ) continue ; 
-      int value = ((const HcalDCCHeader*)(fedData.data()))->getCalibType() ; 
-      calibTypeCounter.at(value)++ ; // increment the counter for this calib type
+
+  //    some inits
+  int numZeroes(0), numPositives(0);
+
+  //    loop over all HCAL FEDs
+  for (int fed=FEDNumbering::MINHCALFEDID;
+       fed<=FEDNumbering::MAXHCALuTCAFEDID; fed++) 
+  {
+      //    skip FEDs in between VME and uTCA    
+      if (fed>FEDNumbering::MAXHCALFEDID && fed<FEDNumbering::MINHCALuTCAFEDID)
+            continue;
+
+      //    get raw data and check if there are empty feds
+      const FEDRawData& fedData = rawdata->FEDData(fed) ; 
+      if ( fedData.size() < 24 ) continue ;
+
+      if (fed<=FEDNumbering::MAXHCALFEDID)
+      {
+          //    VME get event type
+          int eventtype = ((const HcalDCCHeader*)(fedData.data()))->getCalibType(); 
+          eventtype==0 ? numZeroes++ : numPositives++;
```

Comment:
I would suggest to use an `if else` construct rather than a ternary operator here and below:
```c++
if (eventtype==0) ++numZeroes else ++numPositives;
```